docs/site/src/docs/advanced/dev-plugin.md

```diff
@@ -0,0 +1,138 @@
+# Develop plugin.
+
+## Motivations
+
+Sealer support common plugins such as hostname plugin,label plugin,which is build in,user could define and use it
+according their requests. Sealer also support to load out of tree plugin which is written by golang. This page is about
+how to extend the new plugin type and how to develop an out of tree plugin.
+
+## Uses case
+
+### How to develop an out of tree plugin
+
+if user doesn't want their plugin code to be open sourced, we can develop an out of tree plugin to use it.
+
+1. implement the golang plugin interface and expose the variable named `Plugin`.
+
+```shell
+type RunPlugin interface {
```

Comment:
type Interface interface {
	Run(context Context, phase Phase) error
}

Don't change this interface.

Reply:
https://github.com/alibaba/sealer/blob/main/plugin/plugin.go   intree and out of tree has the same interface